outer.c

```diff
@@ -330,6 +343,12 @@ void outer_helper_spawn(struct outer_helper *helper)
 		helper->fd  = fdpair[SOCKET_PARENT];
 		return;
 	}
+
+	/* Make sure all file descriptors except for the ones we're actually using
+	   get closed. This avoids keeping around file descriptors on which
+	   the parent process might be waiting on. */
+	REBIND_FDS_AND_CLOSE(3, fdpair[SOCKET_CHILD]);
```

Comment:
Is there any scenario where you wouldn't want to save 0, 1, 2 ? I almost fee llike oyu aught to omit the arg since it has subtlly different behavior as the rest of the args here

Reply:
(by omit the arg I mean to say I think the function should just assume you want to keep 0, 1, and 2)